test/acceptance/client.js

```diff
@@ -612,27 +654,15 @@ describe('The node.js OC client', function(){
         it('should contain the error details', function(){
 
           var expectedRequestWithExtraParams = {
-            url: 'http://localhost:1234',
-            method: 'post',
+            url: 'http://localhost:1234/hello-world/~1.0.0??hi=john',
```

Comment:
That's because expectedResults are actually regular expressions and '?' is a special symbol so we need to escape it. If we have an expected result like 'some ? text' we need to turn it into 'some \\? text' before the final processing (single back slash doesn't work because '\?' equals '?'). However we also run the getRegExpFromJson(x) function on it, which employs JSON.stringify() that in turn will additionally escape every '\' with '\\' and our string becomes 'some \\\\? text' internally.
That's why I decided to escape it with double question mark '??'.
If you are asking if we can escape '?' with '/?' that's possible, too.
Any other ideas are welcome? :-)